### PR TITLE
[db] add migration to alter pat table allow null expirationTime

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

DB migration part of https://github.com/gitpod-io/gitpod/pull/19327, split in case we need to do revert / rollback

## Related Issue(s)

Relates EXP-1084

## How to test

Should be tested in https://github.com/gitpod-io/gitpod/pull/19327

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/gitpod-db/src/typeorm/entity/db-personal-access-token.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export class DBPersonalAccessToken {
     scopes: string[];
 
     @Column("datetime")
-    expirationTime: Date;
+    expirationTime: Date | null;
```
Comment (svenefftinge) [inline components/gitpod-db/src/typeorm/entity/db-personal-access-token.ts:56]: AFAIR we've really never used `null` as value. Wouldn't setting a date in distant future be less likely to cause problems?

Comment (mustard-mh) [reply components/gitpod-db/src/typeorm/entity/db-personal-access-token.ts:56]: @svenefftinge So just add option to allow 5 years? And not show `No expires!` but actual expiration date on dashboard?

SCR-20240117-dmpv

Comment (mustard-mh) [reply components/gitpod-db/src/typeorm/entity/db-personal-access-token.ts:56]: Or hard code something like `2099-01-01 00:00:00` as never expires

**Comment by mustard-mh** [comment]
Close as we don't need to modify db https://github.com/gitpod-io/gitpod/pull/19328#discussion_r1454741950

**Pull request closed without merging**